commerce-cloud-code/app_yourshophere/cartridge/controllers/Checkout.js

```diff
@@ -0,0 +1,150 @@
+
+const server = require('server');
+
+server.get('Show', (req, res, next) => {
+    const hxPartial = request.httpParameterMap.hxpartial;
```

Comment:
I think it would be great to introduce some checkout manager class which handles the state and flow. Is a step complete, does it have errors and needs to be shown again, did the user want to make edits etc.

Reply:
Yes, thank you. That became clear when working a few minutes yesterday. 

What I ultimately want to achieve is a setup where steps and flow doesn't really matter as much and a google doc like behaviour where you can start punching in stuff, leave the form half done and invalid. You can add more to the basket at anytime and don't loose progress.
That said, as per your point a singular object per user to track what they achieved and whats left to do would be very helpful